### PR TITLE
Addhost tests

### PR DESCRIPTION
**Pull request by nimrof**
Added tests for addhost.php cli tool and addhost function.
Made some changes to addhost and createhost function in order to make addhost run without error when the additional parameter has default value 'array()'

There are some test in addhostCliTest.php for $device->authname that is commented out,
the tests fail, but if i read the help text correct, they should work, but i might be wrong

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
I'd like to see addhost.php go away in favor of `./lnms device:add`

File: tests/addhostCliTest.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+    public function tearDown() : void
+    {
+        exec("./delhost.php ".$this->hostName." ");
+        parent::tearDown();
+    }
+
+    public function testCLIsnmpV1()
+    {
+        exec("./addhost.php -g poller_group -f -p ifIndex ".$this->hostName." community v1 111 tcp");
```
Comment (murrant) [inline tests/addhostCliTest.php:64]: This should be `\Artisan::call('device:add', ...)`

**Comment by murrant** [comment]
I really appreciate all the tests!

File: tests/addhostCliTest.php (hunk 1 above)
Comment (nimrof) [reply tests/addhostCliTest.php:64]: Sorry for late reply.
I will do the changes

File: tests/addhostCliTest.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            $this->assertNotNull($device);
+
+            $this->assertEquals($mode, $device->transport, "Wrong snmp transport (udp/tcp) ipv4/ipv6");
+            exec("./delhost.php ".$host." ");
+        }
+    }
+
+    public function testSnmpV3AuthProtocol()
+    {
+//        $modes = array('md5', 'sha', 'sha-512', 'sha-384', 'sha-256', 'sha-224');
```
Comment (nimrof) [inline tests/addhostCliTest.php:113]: I commented out sha-512, sha-384, sha-256, sha-224 as the test will fail

**Comment by nimrof** [comment]
@murrant 
Hi,
Tried using device:remove, but it printed some text to the console, so i did not use it.
I can always add a quiet -q mode to device:remove to fix it

**Comment by murrant** [comment]
The problem is device:remove passes through the output of addhost.php rather than using the standard output functions, so -q won't work until it is converted to remove the device directly instead of being a passthrough.

**Comment by murrant** [comment]
oh, it isn't using the testing database to remove the devices...

**Comment by murrant** [comment]
Ok, this should be good.  remove is not needed as the transaction is rolled back on teardown.

That is why it always said host not found.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-63-release-changelog-april-2020/11828/1

**Pull request merged**